finality-aleph/src/block/mod.rs

```diff
@@ -256,3 +256,18 @@ pub trait BestBlockSelector<H: Header>: Sync + Send + Clone {
     /// Return header of the leaf that should be considered the best block.
     async fn select_best(&self) -> Result<H, Self::Error>;
 }
+
+pub trait TreePathAnalyzer: Sync {
+    type Error: Display + Debug;
+
+    fn lowest_common_ancestor(&self, a: &BlockId, b: &BlockId) -> Result<BlockId, Self::Error>;
```

Comment:
I'm not quite clear as to why this is the trait in the first place.

Anyway, there is a comment in another file that would result in this being unnecessary, but otherwise I would like to know why this is a trait. Probably in some tests, but then maybe it's better to directly define the below function only?

Reply:
Removed trait and moved impl to `SubstrateChainState`